domain-models-runtime/src/test/java/org/folio/rest/persist/PostgresClientIT.java

```diff
@@ -1433,4 +1446,33 @@ public void selectStreamTxFailed(TestContext context) {
   public void selectStreamParamTxFailed(TestContext context) {
     postgresClient().selectStream(Future.failedFuture("failed"), "SELECT 1", new JsonArray(), context.asyncAssertFailure());
   }
+
+  @Test
+  public void selectDistinctOn(TestContext context) {
+    Async async = context.async();
+    postgresClient = createA(context, TENANT);
+    fillTableWithNumbers(context, postgresClient, TENANT, 5, 4, 5, 6, 5);
+
+    postgresClient.select("SELECT DISTINCT ON (i) i FROM a ORDER BY i", select -> {
+      String result = select.result().getRows().stream()
+        .map(x -> x.getInteger("i").toString())
+        .reduce((n1, n2) -> n1 + ", " + n2)
+        .get();
+      context.assertEquals("4, 5, 6", result);
+      async.complete();
+    });
+    async.awaitSuccess();
+  }
+
+  @Test
+  public void closeAllClientsTest (TestContext testContext) throws Exception {
```

Comment:
This is not related to this distinct on issue and should go into a separate pull request to be discussed and reviewed separately.

Reply:
removed from this PR